python/igraphalg.py

```diff
@@ -0,0 +1,168 @@
+from typing import List
+
+import mgp
+
+from mgp_igraph import MemgraphIgraph
+
+
+@mgp.read_proc
+def maxflow(
+    ctx: mgp.ProcCtx,
+    source: mgp.Vertex,
+    destination: mgp.Vertex,
+    capacity: str = "weight",
+) -> mgp.Record(max_flow=mgp.Number):
+
+    graph = MemgraphIgraph(ctx=ctx, directed=True)
+    max_flow = graph.maxflow(source=source, destination=destination, capacity=capacity)
+
+    return mgp.Record(max_flow=max_flow.value)
+
+
+@mgp.read_proc
+def pagerank(
+    ctx: mgp.ProcCtx,
+    damping: mgp.Number = 0.85,
+    niter: int = 100,
+    eps: mgp.Number = 1e-06,
+    weights: mgp.Nullable[str] = None,
+) -> mgp.Record(node=mgp.Vertex, rank=float):
+    graph = MemgraphIgraph(ctx=ctx, directed=True)
+    ranks = graph.pagerank(
+        weights=weights, directed=True, niter=niter, damping=damping, eps=eps
+    )
+
+    return [
+        mgp.Record(node=graph.get_vertex_by_id(node_id), rank=rank)
+        for node_id, rank in enumerate(ranks)
+    ]
+
+
+@mgp.read_proc
+def all_simple_paths(
+    ctx: mgp.ProcCtx,
+    v: mgp.Vertex,
+    to: mgp.Vertex,
+    cutoff: int = -1,
+) -> mgp.Record(paths=mgp.List[mgp.List[mgp.Vertex]]):
+    graph = MemgraphIgraph(ctx=ctx, directed=True)
+
+    return mgp.Record(paths=graph.get_all_simple_paths(v=v, to=to, cutoff=cutoff))
+
+
+@mgp.read_proc
+def min_cut(
+    ctx: mgp.ProcCtx,
+    source: mgp.Vertex,
+    target: mgp.Vertex,
+    capacity: mgp.Nullable[str] = None,
+) -> mgp.Record(partition_vertices=List[List[mgp.Vertex]], value=float):
+    graph = MemgraphIgraph(ctx=ctx, directed=True)
+
+    cut = graph.mincut(source=source, target=target, capacity=capacity)
+
+    return mgp.Record(
+        partition_vertices=[
+            graph.convert_vertex_ids_to_mgp_vertices(vertex_ids=partition)
+            for partition in cut.partition
+        ],
+        value=cut.value,
+    )
+
+
+@mgp.read_proc
+def topological_sort(
+    ctx: mgp.ProcCtx, directed: bool = True, mode: str = "out"
+) -> mgp.Record(nodes=mgp.Nullable[mgp.List[mgp.Vertex]]):
+    graph = MemgraphIgraph(ctx=ctx, directed=directed)
+    sorted_nodes = graph.topological_sort(mode=mode)
```

Comment:
Do we want to check here if the user passed the correct mode?

Reply:
@antoniofilipovic  This looks fixed